src/components/FormFields.tsx

```diff
@@ -36,42 +41,51 @@ export const TextField = createRestyleComponent<Props, Theme>(
   TextInput,
 )
 
-type LabeledTextField = {
+type FormFieldProps = BoxProps<Theme> & {
   label: string
-  value: string
+  value: any
   onChange?: (string) => void
   onTap?: () => void
+  labelProps?: TextProps<Theme>
   disabled?: boolean
 }
 
-export const LabeledTextField: React.FC<LabeledTextField> = ({
+export const LabeledTextField: React.FC<FormFieldProps> = ({
   label,
   value,
   onChange,
   onTap,
   disabled,
+  labelProps = {},
   ...props
 }) => (
-  <Box width="100%" py={4}>
-    <Text fontWeight="bold" color="darkGrey" pb={2}>
-      {label}
-    </Text>
+  <Box width="100%" py={4} {...props}>
+    {!!label && (
+      <Text variant="heading3" pb={1} {...labelProps}>
+        {label}
+      </Text>
+    )}
+
     <TextField
       variant="login"
       autoCapitalize="none"
       autoCorrect={false}
       onChangeText={(text) => onChange?.(text)}
-      onTouchEnd={() => onTap?.()}
       value={value}
+      onTouchEnd={() => onTap?.()}
       editable={disabled !== true}
-      {...props}
     />
   </Box>
 )
 
-export const LabeledDateField = ({ label, value, onChange, ...props }) => (
-  <Box width="100%" py={4}>
-    <Text fontWeight="bold" color="darkGrey" pb={2}>
+export const LabeledDateField: React.FC<FormFieldProps> = ({
+  label,
+  value,
+  onChange,
+  ...props
```

Comment:
**Nitpick** I think its usually best to avoid object destructuring for props but happy to leave as is.

Reply:
I've never heard of any issues with ES6 destructing, Have you got anything you can point me too?

Reply:
As discussed earlier, concerns were more related to not defining prop-types or in this case types.